src/components/CommentList/CommentList.tsx

```diff
@@ -1,36 +1,173 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { css } from "emotion";
 
-import { CommentType, ThreadsType } from "../../types";
+import { space, palette } from "@guardian/src-foundations";
+
+import { Pillar, CommentType, UserProfile, ThreadsType } from "../../types";
+import { CommentForm } from "../CommentForm/CommentForm";
 import { Comment } from "../Comment/Comment";
 
 type Props = {
-  comments?: CommentType[];
+  comment: CommentType;
+  pillar: Pillar;
+  shortUrl: string;
+  user?: UserProfile;
+  onAddComment: (commentId: number, body: string, user: UserProfile) => void;
   threads: ThreadsType;
+  replyComment?: CommentType;
```

Comment:
So my personal preference is to not pass around objects. I would prefer a simple api where we pass just the id and, in the case where we need more information, we read it from either `comment` or `comment.responses` - but that's me 🤷‍♂